### ISSUE TITLE
Extra checks with -d not executed with --noEmit

### ISSUE THREAD

**Issue by mhverbakel** [microsoft/TypeScript#13212]
**TypeScript Version:**  2.1.4

**Code**

```ts
class SomeClass {

}

export = new SomeClass();
```

**Actual behavior:**
This code should give an error, as the exported instance is of a private type. The following error is reported when compiling with `-d`, but not without it:
```
error TS4082: Default export of the module has or is using private name 'SomeClass'.
```
The disadvantage of using `-d` is that it generates `.d.ts` files, which should be ignored (because they are auto-generated). I can not add `*.d.ts` to my gitignore, as some of my interfaces are `.d.ts` files.

I tried to resolve this issue by adding `--noEmit` to the compiler options, but this hides the error (as if it was compiled without `-d`).

**Expected behavior:**
Either report the error always, or report it with `-d` and `--noEmit`.

**Comment by RyanCavanaugh** [comment]
No longer errors at all
